jdbc/src/main/java/org/apache/zeppelin/jdbc/security/JDBCSecurityImpl.java

```diff
@@ -46,10 +46,16 @@ public static void createSecureConfiguration(Properties properties,
         conf.set("hadoop.security.authentication", KERBEROS.toString());
         UserGroupInformation.setConfiguration(conf);
         try {
-          UserGroupInformation.loginUserFromKeytab(
-              properties.getProperty("zeppelin.jdbc.principal"),
-              properties.getProperty("zeppelin.jdbc.keytab.location")
-          );
+          // Check TGT before calling login
+          // Ref: https://github.com/apache/hadoop/blob/release-3.0.1-RC1/hadoop-common-project/
+          // hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java#L1232
+          if (!UserGroupInformation.isSecurityEnabled()
+              || UserGroupInformation.getCurrentUser().getAuthenticationMethod() != KERBEROS
+              || !UserGroupInformation.isLoginKeytabBased()) {
```

Comment:
Is the if condition logic correct ? will login keytab when security is not enabled ?

Reply:
The first condition (`case KERBEROS:`)  indicates that user wanted to enable Kerberos, and this second condition (combination of three flags) indicates that it is not done yet.